File: worker/src/analysis/movement.ts
/** Line-movement classification. Port of bot/analysis/movement.py. */
export type Movement = "stable" | "steam" | "sharp" | "drift";

export function classify(openProb: number, currentProb: number): Movement {
  const ad = Math.abs(currentProb - openProb);
  if (ad < 0.03) return "stable";
  if (ad >= 0.1) return "steam";
  if (ad >= 0.05) return "sharp";
  return "drift";
}

File: bot/analysis/movement.py
"""Line-movement classification from probability drift."""
from __future__ import annotations


def classify(open_prob: float, current_prob: float) -> str:
    """Return stable | steam | sharp | drift for a probability move."""
    delta = current_prob - open_prob
    ad = abs(delta)
    if ad < 0.03:
        return "stable"
    if ad >= 0.10:
        return "steam"
    if ad >= 0.05:
        return "sharp"
    return "drift"
